gnss_lib_py/utils/gnss_models.py

```diff
@@ -0,0 +1,658 @@
+    # Loop through the measurement file per time step
+    for gps_millis, _, measure_frame in measurements.loop_time('gps_millis',
+                                                        delta_t_decimals=delta_t_dec):
+        # Sort the satellites
+        rx_ephem, sorted_sats_ind, inv_sort_order = _sort_ephem_measures(measure_frame, ephem)
+        # Create new NavData with SV positions and velocities
+        # If they're not given, the SV states computed with measures will be used
+        try:
+            measure_frame.in_rows(sv_state_rows)
+            use_posvel = False
+            sv_posvel = NavData()
+            for row in sv_state_rows:
+                sv_posvel[row] = measure_frame[row]
+            for row in info_rows:
+                sv_posvel[row] = measure_frame[row]
+            # sv_posvel = sv_posvel.sort(ind=sorted_sats_ind)
+            sv_posvel = sv_posvel.sort(ind=sorted_sats_ind)
+        except KeyError:
+            sv_posvel = None
+            use_posvel = True
+
+        # Extract RX states into State NavData
+        state = NavData()
+        for row in rx_pos_rows:
+            state[row] = measure_frame[row, 0]
+        # velocity and clock rows
+        vel_clk_rows = rx_vel_rows_to_find + rx_clk_rows_to_find
+        for row in vel_clk_rows:
+            try:
+                row_idx = measure_frame.find_wildcard_indexes(row,max_allow=1)
+                state[row_idx[row][0]] = measure_frame[row_idx[row][0], 0]
+            except KeyError:
+                warnings.warn("Assuming 0 "+ row + " for Rx", RuntimeWarning)
+                state[row] = 0
+
+        # Compute measurements
+        if pseudorange or doppler:
+            est_meas, sv_posvel = expected_measures(gps_millis, state,
+                                                    ephem=rx_ephem,
+                                                    sv_posvel=sv_posvel)
+            # Reverse the sorting to match the input measurements
+            est_meas = est_meas.sort(ind=inv_sort_order)
+        else:
+            est_meas = None
+        if corrections:
+            est_clk, est_trp, est_iono = calculate_pseudorange_corr(gps_millis,
+                                        state=state, ephem=rx_ephem,
+                                        sv_posvel=sv_posvel, iono_params=iono_params)
+            # Reverse the sorting to match the input measurements
+            est_clk = est_clk[inv_sort_order]
+            est_trp = est_trp[inv_sort_order]
+            est_iono = est_iono[inv_sort_order]
+        else:
+            est_clk = None
+            est_trp = None
+            est_iono = None
+        # Add required values to new rows
+        if sv_posvel is not None:
+            # Reverse the sorting to match the input measurements
+            sv_posvel = sv_posvel.sort(ind=sorted_sats_ind)
+        est_frame = NavData()
+        if pseudorange:
+            est_frame['est_pr_m'] = est_meas['est_pr_m']
```

Comment:
How does `est_pr_m` compare with `raw_pr_m` and `corr_pr_m`? There's no documentation for it on the reference page. Unclear what I'm supposed to do with `est_pr_m` and how I obtain the needed `corr_pr_m` row that I need for the localization algorithms.